### PR TITLE
bug(trace-viewer): update trace file on trace viewer when test completes

### PR DESCRIPTION
**Pull request by ruifigueira**
Fixes https://github.com/microsoft/playwright/issues/31314

### CODE REVIEW

**Comment by pavelfeldman** [comment]
I don't think the error that you see in console is related to the behavior that you are changing. What's the idea behind this change, could you verbalize it?

**Comment by ruifigueira** [comment]
This change updates the trace displayed in the trace viewer when a test in a multiple test run completes (basically, it replaces the live json trace file, which is being polled during live mode, into the final zip trace file). Otherwise, VSCode `TraceViewer` instance doesn't send the new trace file to `_traceViewerProcess`, which will continue to send `loadTraceRequested` messages via websocket to the traceviewer webapp with the old json trace path.

In my tests, it fixed the issue.

**Comment by ruifigueira** [comment]
BTW, the reason why it works with one single test running and fails with more than one is because `enqueuedSingleTest` is `true`  and so it calls the `showTrace`:

https://github.com/microsoft/playwright-vscode/blob/main/src/extension.ts#L486-L487

The check I added there is just to ensure that the trace viewer is showing the trace for the right test item. If they are the same, the trace viewer must be displaying the old trace value that is now being updated.

File: src/extension.ts [hunk 1]
```diff
@@ -488,8 +488,10 @@ export class Extension implements RunHooks {
           return;
 
         const trace = result.attachments.find(a => a.name === 'trace')?.path || '';
+        const prevTrace = (testItem as any)[traceUrlSymbol];
```
Comment (pavelfeldman) [inline src/extension.ts:491]: Relying on the previous file is a bit fragile, but given that there is no clear way to get currently selected tree item, I'm ok with it. Let's add a comment explaining what is going on.

File: src/traceViewer.ts [hunk 2]
```diff
@@ -38,6 +39,10 @@ export class TraceViewer implements vscodeTypes.Disposable {
     }));
   }
 
+  get currentFile() {
```
Comment (pavelfeldman) [inline src/traceViewer.ts:42]: We don't use getters, should be plain method.

**Review by pavelfeldman — APPROVED** [review]

**Comment by pavelfeldman** [comment]
> Otherwise, VSCode TraceViewer instance doesn't send the new trace file to _traceViewerProcess, which will continue to send loadTraceRequested messages via websocket to the traceviewer webapp with the old json trace path

That's normal behavior for the case of live traces - we race with the trace viewer here, so it is Ok for the files to not be there. I understand that you are removing a case where update is hammering against the missing file, but the JSON error will still be happening in other cases. I'd suggest that you change the title of the PR to reflect the actual change.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**